Source/ARTRealtimeChannel.m

```diff
@@ -1233,13 +1233,19 @@ - (void)map:(ARTPresenceMap *)map didRemovedMemberNoLongerPresent:(ARTPresenceMe
 }
 
 - (void)map:(ARTPresenceMap *)map shouldReenterLocalMember:(ARTPresenceMessage *)presence {
-    [self.presence enterClient:presence.clientId data:presence.data callback:^(ARTErrorInfo *error) {
-        NSString *message = [NSString stringWithFormat:@"Re-entering member \"%@\" as failed with code %ld (%@)", presence.clientId, (long)error.code, error.message];
-        ARTErrorInfo *reenterError = [ARTErrorInfo createWithCode:ARTErrorUnableToAutomaticallyReEnterPresenceChannel message:message];
-        ARTChannelStateChange *stateChange = [[ARTChannelStateChange alloc] initWithCurrent:self.state_nosync previous:self.state_nosync event:ARTChannelEventUpdate reason:reenterError resumed:true];
-        [self emit:stateChange.event with:stateChange];
+    [self.presence enterWithPresenceMessageId:presence.id clientId:presence.clientId data:presence.data callback:^(ARTErrorInfo *error) {
+        if (error != nil) {
```

Comment:
https://github.com/ably/ably-cocoa/pull/1795

Reply:
@maratal can you check this comment? This will cause refactoring at multiple places.

Reply:
Was there a discussion with realtime team about this? Seems like it should go into the spec first (removing `id`).

Reply:
@sacOO7 Marat is right. The spec [does say](https://sdk.ably.com/builds/ably/specification/main/features/#RTP17g) you should send the id; if you think that's buggy and needs to be changed, please open an issue in the specification repo, preferably with repro instructions for a test that fails if you use the id. That way if it is wrong, we can fix it in the spec rather than relying on you telling all the other sdk developers to ignore that spec item :slightly_smiling_face: 

(from memory, the purpose of sending the original id is so that if the user sends an explicit leave or update immediately after reattaching, that will trump the automatic re-enter in the newness check. possibly we need to come up with a solution that has that property while fixing whatever issue Sachin is encountering)

Reply:
Yeah, we need to check on this one. Because we are not sending `id` in either `ably-dotnet` or `ably-go` for automatic re-enter.

Reply:
Also @maratal can you add test for this?

Reply:
Can you please add test for this

Reply:
I am resolving this discussion for now